orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -558,6 +558,49 @@ void resize_impl_ort_(
   return;
 }
 
+// The following was copied from onnxruntime/core/providers/cuda/math/binary_elementwise_ops.cc
+// This method compute the resulting shape from broadcasting.
+Status ComputeOutputShape(
+  const std::string& node_name,
+  const onnxruntime::TensorShape& lhs_shape,
+  const onnxruntime::TensorShape& rhs_shape,
+  onnxruntime::TensorShape& out_shape) {
+  size_t lhs_rank = lhs_shape.NumDimensions();
+  size_t rhs_rank = rhs_shape.NumDimensions();
+  size_t out_rank = std::max(lhs_rank, rhs_rank);
+
+  std::vector<int64_t> output_dims(out_rank, 0);
+  for (size_t i = 0; i < out_rank; ++i) {
+    int64_t lhs_dim = 1;
+    if (i < lhs_rank)
+      lhs_dim = lhs_shape[lhs_rank - 1 - i];
+    int64_t rhs_dim = 1;
+    if (i < rhs_rank)
+      rhs_dim = rhs_shape[rhs_rank - 1 - i];
+    int64_t max = std::max(lhs_dim, rhs_dim);
+    int64_t min = std::min(lhs_dim, rhs_dim);
+    int64_t out_dim = (min == 0 ? min : max);  // special case a dim value of 0.
+    if (lhs_dim != out_dim && lhs_dim != 1)
+      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, node_name, ": left operand cannot broadcast on dim ", lhs_rank - 1 - i,
```

Comment:
I don't see the value in requiring node_name as a param just for the error message.

Reply:
it is part of ort_make_status so I believe I need something there. Open to suggestions of strings to put there so as not to pass it in.

Reply:
It's not required, just concatenated for convenience.

from include/onnxruntime/core/common/common.h

#define ORT_MAKE_STATUS(category, code, ...)                     \
  ::onnxruntime::common::Status(::onnxruntime::common::category, \
                                ::onnxruntime::common::code,     \
                                ::onnxruntime::MakeString(__VA_ARGS__))